object/src/mem.rs

```diff
@@ -336,6 +336,19 @@ where
         self.entries.insert(elt.tag(), elt)
     }
 
+    /// Removes and returns a particular DICOM element by its tag.
+    pub fn take_element(&mut self, tag: Tag) -> Result<InMemElement<D>> {
+        self.entries
+            .remove(&tag)
+            .context(NoSuchDataElementTag { tag })
+    }
+
+    /// Removes and returns a particular DICOM element by its name.
+    pub fn take_element_by_name(&mut self, name: &str) -> Result<InMemElement<D>> {
+        let tag = self.lookup_name(name)?;
+        self.take_element(tag)
```

Comment:
Hmm, I have reminded myself on why I chose to replicate the implementation of `element` and `element_by_name`, instead of just delegating one to the other. If `take_element` fails, it returns an error of variant `NoSuchElementTag`, which is less informative than the variant `NoSuchElementAlias`.

Please see how it was done in `element_by_name` and ensure that the error returned contains the attribute alias. Let me know if further clarification is needed.

Reply:
That makes sense, thanks for the feedback! I'll update that